exploration/0001-builtin-registry-capabilities.md

```diff
@@ -0,0 +1,88 @@
+# Capabilities for built-in registry
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@ryzokuken</dd>
+		<dt>First proposed</dt>
+		<dd>2023-08-22</dd>
+		<dt>Pull Request</dt>
+		<dd><a href="https://github.com/unicode-org/message-format-wg/pull/457">#457</a></dd>
+	</dl>
+</details>
+
+## Objective
+
+Document the set of capabilities needed by the _built-in registry_ in order to provide a reasonably complete set of
+functions.
+This includes ensuring sufficient compatibility with the various functions available in ICU4X, JavaScript `Intl` as well
+as MessageFormat v1.
+
+## Background
+
+Users have the ability to specify and call arbitrary custom functions in their messages thanks to the function registry.
+That said, there are a number of common i18n operations that are quite universal and there's a lot of value in
+standardizing the usage of these common operations across the ecosystem.
+This document aims to list down the basic capabilities that implementations need to be provide in a _built-in registry_.
+
+## Use-Cases
+
+From formatting date and time values to performing selection based on plural rules, the use cases for a minimal set of
+common tools standardized across implementations are virtually everywhere.
+While a number of applications might require more complex functions with custom logic, a well-provisioned default
+registry could significantly lower the barrier of entry as well as the need to write and maintain many custom functions.
+
+## Requirements
+
+Implementations are required to implement and expose the entire _built-in registry_ that would provide the capabilities
+in this document.
```

Comment:
The requirements section should be requirements on this design, rather than what this design requires. A comprehensive set of requirements could be rather extensive, but it's still a good idea to spend the effort. One way to do that is to populate this section as you write the design.